File: Integral_process_demo/server/src/main/java/com/yzq/Integral_process_server/server/RegisterServlet.java
package com.yzq.Integral_process_server.server;

import com.alibaba.druid.pool.DruidPooledConnection;
import com.yzq.Integral_process_server.server.utils.DruidUtils;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.sql.PreparedStatement;
import java.sql.SQLException;


/**
 * @author : yuzhiqiang (zhiqiang.yu.xeon@gmail.com)
 * @description: 注册  (往数据库增加一条数据)
 * @date : 2021/4/4
 * @time : 14:28
 */
@WebServlet(name = "RegisterServlet", value = "/register")
public class RegisterServlet extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {


    }

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

        String account = request.getParameter("account");
        System.out.println("account = " + account);
        String pwd = request.getParameter("pwd");
        System.out.println("pwd = " + pwd);


        /*插入到数据库*/

        try {
            DruidPooledConnection connection = DruidUtils.druidDataSource.getConnection();
            String sql = "insert into account(user_name,pwd) values(?,?)";
            PreparedStatement preparedStatement = connection.prepareStatement(sql);
            preparedStatement.setString(1, account);
            preparedStatement.setString(2, pwd);

            int i = preparedStatement.executeUpdate();

            if (i > 0) {
                response.getWriter().write("成功");
            } else {
                response.getWriter().write("失败");
            }
            connection.close();

        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
    }
}

File: Integral_process_demo/kotlinServer/src/main/kotlin/com/yzq/kotlin_server/kotlinServer/servlet/RegisterServlet.kt
package com.yzq.kotlin_server.kotlinServer.servlet

import com.yzq.kotlin_server.kotlinServer.utils.DruidUtils
import java.lang.Exception
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse


/**
 * @description: 注册接口 （对应数据库中的 增）
 * @author : yuzhiqiang (zhiqiang.yu.xeon@gmail.com)
 * @date   : 2021/4/4
 * @time   : 20:30
 */

@WebServlet(name = "RegisterServlet", value = ["/register"])
class RegisterServlet : HttpServlet() {

    override fun doPost(request: HttpServletRequest, response: HttpServletResponse) {

        val account = request.getParameter("account")
        println("account = ${account}")
        val pwd = request.getParameter("pwd")
        println("pwd = ${pwd}")


        val sql = "insert into account(user_name,pwd) values(?,?)"


        try {
            val connection = DruidUtils.druidDataSource.getConnection()
            val prepareStatement = connection.prepareStatement(sql)
            prepareStatement.setString(1, account)
            prepareStatement.setString(2, pwd)

            val executeUpdate = prepareStatement.executeUpdate()

            println("executeUpdate = ${executeUpdate}")

            connection.close()

            if (executeUpdate > 0) {
                response.writer.write("注册成功")
            } else {
                response.writer.write("注册失败")
            }

        } catch (e: Exception) {
            e.printStackTrace()
            response.writer.write("注册失败")
        } finally {


        }


    }
}